src/main/java/org/axion/axion_api_gateway/RequestValidator.java

```diff
@@ -0,0 +1,68 @@
+package org.axion.axion_api_gateway;
+
+import com.typesafe.config.Config;
+import jakarta.servlet.http.HttpServletRequest;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.axion.axion_api_gateway.config.AppConfig;
+import org.axion.axion_api_gateway.config.ConfigNotFoundException;
+import org.axion.axion_api_gateway.config.ServiceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RequestValidator {
+
+    private final AppConfig appConfig;
+    public RequestValidator(AppConfig appConfig) {
+        this.appConfig = appConfig;
+    }
+
+    public static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+
+    private static final Logger log = LoggerFactory.getLogger(RequestValidator.class);
+
+    public void checkRequestViolations(HttpServletRequest request) {
+        String header = request.getHeader("Authorization");
+        if (header == null || header.isEmpty()) {
+            throw new RequestValidationException("Authorization header is missing or empty", HttpStatus.BAD_REQUEST);
+        }
+
+        String token = header.substring("Bearer ".length());
+
+        boolean isTokenValid = validateTokenWithAuthService(token);
+        if (!isTokenValid) {
+            throw new RequestValidationException("Invalid token", HttpStatus.UNAUTHENTICATED);
+        }
+    }
+
+    public String createJson(String token) {
```

Comment:
This method should be private, it's only for this class context.

Reply:
FIxed